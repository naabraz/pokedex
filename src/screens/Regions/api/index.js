import {
  headers,
  url,
} from 'config'

export const fetchRegions = () =>
  fetch(`${url}/region`, {
    headers,
  })
    .then(res => res.json())
    .then(data => data)
    .catch((e) => new Error(e))
