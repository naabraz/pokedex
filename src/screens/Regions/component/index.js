import React from 'react'
import PropTypes from 'prop-types'

import Wrapper from 'components/Wrapper'
import {
  RegionsList,
  RegionName,
  RegionIcon,
  RegionView,
  RegionButton,
} from '../style'
import { worldwide } from 'assets/icons'
import { capitalize } from 'helpers'

const Regions = (props) => {
  const { regions, navigation } = props

  return (
    <Wrapper>
      <RegionsList
        data={regions}
        numColumns={3}
        renderItem={({ item }) =>
          <RegionView>
            <RegionButton onPress={() => navigation.navigate('Pokedex', { detailURL: item.url })}>
              <RegionIcon source={worldwide} />
              <RegionName>
                {capitalize(item.name)}
              </RegionName>
            </RegionButton>
          </RegionView>}
        keyExtractor={({ name }) => name} />
    </Wrapper>
  )
}

Regions.propTypes = {
  regions: PropTypes.array,
  navigation: PropTypes.object,
}

export default Regions
