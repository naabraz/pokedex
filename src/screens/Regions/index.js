import RegionsContainer from './container'

export default RegionsContainer
