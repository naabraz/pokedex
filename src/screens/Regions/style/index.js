import styled from 'styled-components/native'

import {
  whiteSmoke,
  black,
} from 'themes'

export const RegionsList = styled.FlatList`
`

export const RegionView = styled.View`
  background-color: ${whiteSmoke}
  border-radius: 5
  padding-top: 20px
  padding-bottom: 20px
  padding-left: 50px
  padding-right: 50px
  margin: 10px
`

export const RegionButton = styled.TouchableOpacity`
  align-items: center
`

export const RegionName = styled.Text`
  color: ${black}
  font-size: 24
`

export const RegionIcon = styled.Image`
  width: 64
  height: 64
`
