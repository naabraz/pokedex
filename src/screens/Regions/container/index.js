// @flow

import React, { Component } from 'react'

import { fetchRegions } from '../api'
import Regions from '../component'

type IRegionsContainer = {
  navigation: Object,
}

type State = {
  regions: Array<Object>
}

class RegionsContainer extends Component<IRegionsContainer, State> {
  state = {
    regions: [],
  }

  async componentDidMount () {
    const { results } = await fetchRegions()
    this.setState({ regions: results })
  }

  render () {
    const { regions } = this.state
    const { navigation } = this.props

    return (
      <Regions regions={regions} navigation={navigation} />
    )
  }
}

export default RegionsContainer
