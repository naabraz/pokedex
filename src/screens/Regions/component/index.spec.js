import React from 'react'
import renderer from 'react-test-renderer'

import Regions from '.'

describe('Regions component test', () => {
  it('Should render component correctly', () => {
    const regionsList = [{ name: 'kantoo' }, { name: 'johto' }]
    const regions = renderer.create(<Regions regions={regionsList} />).toJSON()
    expect(regions).toMatchSnapshot()
  })
})
