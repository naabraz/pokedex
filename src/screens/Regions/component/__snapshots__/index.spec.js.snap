// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Regions component test Should render component correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#e32722",
        "display": "flex",
        "height": "100%",
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#64afe1",
          "borderRadius": 5,
          "height": "85%",
          "justifyContent": "center",
          "width": "85%",
        },
      ]
    }
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "name": "kantoo",
          },
          Object {
            "name": "johto",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Array [
          Object {},
        ]
      }
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#e6e6e6",
                  "borderRadius": 5,
                  "marginBottom": 10,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginTop": 10,
                  "paddingBottom": 20,
                  "paddingLeft": 50,
                  "paddingRight": 50,
                  "paddingTop": 20,
                },
              ]
            }
          >
            <View
              accessible={true}
              clickable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 24,
                    },
                  ]
                }
              >
                Kantoo
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#e6e6e6",
                  "borderRadius": 5,
                  "marginBottom": 10,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginTop": 10,
                  "paddingBottom": 20,
                  "paddingLeft": 50,
                  "paddingRight": 50,
                  "paddingTop": 20,
                },
              ]
            }
          >
            <View
              accessible={true}
              clickable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 24,
                    },
                  ]
                }
              >
                Johto
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
